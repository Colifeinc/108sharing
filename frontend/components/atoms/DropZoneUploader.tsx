import * as React from 'react'
import { DropEvent, useDropzone } from 'react-dropzone'
import I18n from 'core/i18n'
import { UPLOAD_FILE_MAX_SIZE } from 'constants/constants'
import classNames from 'classnames'

interface IProps {
  className?: string
  onDrop(acceptedFiles, rejectedFiles, event: DropEvent): Promise<void> | void
}

const DropZoneUploader = React.memo<IProps>(({ className = '', onDrop }) => {
  const {
    isDragActive,
    getRootProps,
    getInputProps,
    isDragReject,
    rejectedFiles,
    rootRef,
  } = useDropzone({
    accept: 'image/*',
    maxSize: UPLOAD_FILE_MAX_SIZE,
    minSize: 0,
    onDrop: async (acceptedFiles, rejectedFiles, event) => {
      await onDrop(acceptedFiles, rejectedFiles, event)
      if (rootRef?.current?.blur) rootRef?.current?.blur()
    },
  })
  const isFileTooLarge = rejectedFiles.length > 0 && rejectedFiles[0].size > UPLOAD_FILE_MAX_SIZE
  const active = isDragActive && !isDragReject
  const rejected = isDragReject || isFileTooLarge

  return (
    <div
      {...getRootProps()}
      className={classNames([
        'flex items-center justify-center p-2 border-dashed border-2 rounded-lg text-sm font-bold cursor-pointer text-center',
        'transition duration-150 focus:border-primary focus:outline-none',
        'file-drop-input',
        !active && !rejected && 'border-neutral-300 text-neutral-600',
        active && 'border-primary text-primary',
        rejected && 'border-red text-red',
        className,
      ])}
    >
      <input {...getInputProps()} />
      <p className="file-drop-input-text">
        {!isDragActive && I18n.t('dropzone.select_or_drag_and_drop')}
        {isDragActive && !isDragReject && I18n.t('dropzone.drop')}
        {isDragReject && I18n.t('dropzone.invalid_file')}
        {isFileTooLarge && I18n.t('dropzone.too_large')}
      </p>
    </div>
  )
})

export default DropZoneUploader
