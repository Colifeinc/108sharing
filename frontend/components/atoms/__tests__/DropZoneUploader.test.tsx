import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'
import DropZoneUploader from '../DropZoneUploader'

jest.mock('../../../core/i18n')

describe('atoms/filters/DropZoneUploader.tsx', () => {
  const onDrop = jest.fn()
  const props = {
    onDrop,
  }

  it('should matches to the snapshot', () => {
    const wrapper = shallow(<DropZoneUploader {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
