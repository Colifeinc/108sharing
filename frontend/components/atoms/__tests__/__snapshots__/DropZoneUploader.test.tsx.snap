// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/filters/DropZoneUploader.tsx should matches to the snapshot 1`] = `
<div
  className="flex items-center justify-center p-2 border-dashed border-2 rounded-lg text-sm font-bold cursor-pointer text-center transition duration-150 focus:border-primary focus:outline-none file-drop-input border-neutral-300 text-neutral-600"
  onBlur={[Function]}
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <input
    accept="image/*"
    autoComplete="off"
    multiple={true}
    onChange={[Function]}
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex={-1}
    type="file"
  />
  <p
    className="file-drop-input-text"
  >
    Drag 'n' drop some files here, or click to select files
  </p>
</div>
`;
